docs/src/resources/two-side-column.md

```diff
@@ -0,0 +1,712 @@
+```
+</td>
+         </tr>
+  <tr>
+<td>
+
+```
+module CaseStatementEnumModule2 (input clk);
+ 
+  typedef enum {INIT, IDLE, START, READY} t_state;
+    t_state state;
+    
+    
+ 
+    always @(posedge clk) begin
+        case (state)
+            IDLE    : state = START;
+            START   : state = READY;
+            READY   : state = IDLE ;
+            default : state = IDLE ;
+        endcase
+    end
+    
+endmodule
+```
+</td>
+<td>
+
+```scala mdoc:silent
+class CaseStatementEnumModule2 extends Module {
+  
+  object AluMux1Sel extends ChiselEnum {
+   val INIT  = Value(0x03.U) 
+    val IDLE  = Value(0x13.U) 
+    val START = Value(0x17.U) 
+    val READY = Value(0x23.U) 
+  }
+   import AluMux1Sel._
+    val state = RegInit(INIT)
+  val nextState = IO(Output(AluMux1Sel()))
+  nextState := state
+
+  
+  switch (state) {
+  is (INIT) {
+   state := IDLE   
+  }
+  is (IDLE) {  
+   state := START
+  }
+  is (START) { 
+   state := READY
+  }
+ is (READY) { 
+   state := IDLE
+  }
+  }
+}
+```
+```scala mdoc:invisible
+ChiselStage.emitVerilog(new CaseStatementEnumModule2)
+```
+</td>
+         </tr>
+    </table>
+<html>
+<body>
+
+<!--
+# SystemVerilog Interfaces
+
+<html>
+<body>
+    <table border ="0">
+          <tr>
+            <td><b style="font-size:30px">Verilog</b></td>
+            <td><b style="font-size:30px">Chisel</b></td>
+         </tr>
+         <tr>
+<td>
+
+```
+module MyInterfaceModule(
+  input        clock,
+  input        reset,
+  output       io_in_ready,
+  input        io_in_valid,
+  input  [7:0] io_in_bits,
+  input        io_out_ready,
+  output       io_out_valid,
+  output [7:0] io_out_bits
+);
+  assign io_in_ready = io_out_ready; // @[main.scala 17:12]
+  assign io_out_valid = io_in_valid; // @[main.scala 17:12]
+  assign io_out_bits = io_in_bits; // @[main.scala 17:12]
+endmodule
+```
+</td>
+<td>
+
+```scala mdoc:silent
+class MyInterfaceModule extends Module {
+val io = IO(new Bundle {
+val in = Flipped(DecoupledIO(UInt(8.W)))
+val out = DecoupledIO(UInt(8.W))
+})
+
+val tmp = Wire(DecoupledIO(UInt(8.W)))
+tmp <> io.in
+io.out <> tmp
+io.out <> io.in
+}
+```
+```scala mdoc:invisible
+ChiselStage.emitVerilog(new MyInterfaceModule)
+```
+
+</td>
+         </tr>
+    </table>
+
+<html>
+<body>
+-->
+
+
+# Memory Modules
+
+<html>
+<body>
+    <table border ="0">
+          <tr>
+            <td><b style="font-size:30px">Verilog</b></td>
+            <td><b style="font-size:30px">Chisel</b></td>
+         </tr>
+         <tr>
+<td>
+
+```
+module ReadWriteMem(
+input         clock,
+input         io_enable,
+input         io_write,
+input  [9:0]  io_addr,
+input  [31:0] io_dataIn,
+output [31:0] io_dataOut
+);
+
+reg [31:0] mem [0:1023];
+
+assign io_dataOut = mem[io_addr];
+
+always @(posedge clock) begin
+  if (io_enable && io_write) begin
+  mem[io_addr] <= io_dataIn;
+end
+
+endmodule
+```
+</td>
+
+<td>
+
+```scala mdoc:silent
+class ReadWriteSmem extends Module {
```

Comment:
Let's put this chisel/verilog for ReadWriteSmem into its own row from ReadWriteMem

Reply:
Here is the verilog for ReadWriteSmem:

```
module ReadWriteSmem(
  input         clock,
  input         reset,
  input         io_enable,
  input         io_write,
  input  [9:0]  io_addr,
  input  [31:0] io_dataIn,
  output [31:0] io_dataOut
);

  reg [31:0] mem [0:1023]; // @[main.scala 15:24]

  reg [9:0] addr_delay;

  assign io_dataOut = mem[addr_delay]

  always @(posedge clock) begin
    if (io_enable & io_write) begin
      mem[io_addr] <= io_data;
    end
    if (io_enable) begin
      addr_delay <= io_addr;
    end
  end
endmodule
```